pcu/PCU.cc

```diff
@@ -52,6 +52,23 @@
 #include <time.h>
 namespace pcu {
 
+int PCU_Init(int *argc, char ***argv) {
+#ifndef SCOREC_NO_MPI
+  return MPI_Init(argc, argv);
+#else
+  (void) argc, (void) argv;
+  return 0;
+#endif
+}
+
+int PCU_Finalize() {
+#ifndef SCOREC_NO_MPI
+  return MPI_Finalize();
```

Comment:
As above, I think we should check if MPI was already finalized before calling finalize ourselves.  https://www.mpich.org/static/docs/latest/www3/MPI_Finalized.html